presto-verifier/src/main/java/com/facebook/presto/verifier/QueryRewriter.java

```diff
@@ -0,0 +1,153 @@
+            throws SQLException
+    {
+        List<String> parts = new ArrayList<>();
+        int originalSize = statement.getName().getOriginalParts().size();
+        int prefixSize = rewritePrefix.getOriginalParts().size();
+        if (originalSize > prefixSize) {
+            parts.addAll(statement.getName().getOriginalParts().subList(0, originalSize - prefixSize));
+        }
+        parts.addAll(rewritePrefix.getOriginalParts());
+        parts.set(parts.size() - 1, rewritePrefix.getSuffix() + UUID.randomUUID().toString().replace("-", ""));
+        QualifiedName temporaryTableName = QualifiedName.of(parts);
+        Statement rewritten = new CreateTableAsSelect(temporaryTableName, statement.getQuery(), statement.isNotExists(), statement.getProperties(), statement.isWithData());
+        String createTableAsSql = formatSql(rewritten);
+        String checksumSql = checksumSql(metaData, catalogOverride.orElse(query.getCatalog()), schemaOverride.orElse(query.getSchema()), statement.getName().getSuffix(), temporaryTableName);
+        String dropTableSql = dropTableSql(temporaryTableName);
+        return new Query(query.getCatalog(), query.getSchema(), ImmutableList.of(createTableAsSql), checksumSql, ImmutableList.of(dropTableSql), query.getUsername(), query.getPassword(), query.getSessionProperties());
+    }
+
+    private String checksumSql(DatabaseMetaData metaData, String catalog, String schema, String existingTable, QualifiedName table)
+            throws SQLException
+    {
+        ResultSet columns = metaData.getColumns(catalog, escapeNamePattern(metaData.getSearchStringEscape(), schema), escapeNamePattern(metaData.getSearchStringEscape(), existingTable), null);
+
+        ImmutableList.Builder<SelectItem> selectItems = ImmutableList.builder();
+        while (columns.next()) {
+            String name = columns.getString("COLUMN_NAME");
+            int type = columns.getInt("DATA_TYPE");
+            Expression expression = new QualifiedNameReference(QualifiedName.of(name));
+            if (type == Types.REAL || type == Types.DOUBLE || type == Types.FLOAT) {
+                expression = new FunctionCall(QualifiedName.of("round"), ImmutableList.of(expression, new LongLiteral(Integer.toString(doublePrecision))));
+            }
+            selectItems.add(new SingleColumn(new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(expression))));
+        }
+        Select select = new Select(false, selectItems.build());
+        return formatSql(new QuerySpecification(select, Optional.of(new Table(table)), Optional.empty(), Optional.empty(), Optional.empty(), ImmutableList.of(), Optional.empty()));
+    }
+
+    private static String escapeNamePattern(String escapeSequence, String name)
+    {
+        checkArgument(!escapeSequence.equals("%") && !escapeSequence.equals("_"), "Cannot use % or _ as an escape sequence.");
+        return name.replace("%", escapeSequence + "%").replace("_", escapeSequence + "_");
```

Comment:
This might be easier to read like
```java
return name
        .replace("%", escapeSequence + "%")
        .replace("_", escapeSequence + "_");